axelrod/strategies/memorytwo.py

```diff
@@ -1,11 +1,203 @@
+        if self.name == 'Generic Memory Two Player':
+            self.name = "%s: %s" % (self.name, sixteen_vector)
+
+    def set_sixteen_vector(self, sixteen_vector: Tuple):
+        if not all(0 <= p <= 1 for p in sixteen_vector):
+            raise ValueError("An element in the probability vector, {}, is not "
+                             "between 0 and 1.".format(str(sixteen_vector)))
+
+        states = [(hist[:2], hist[2:])
+                  for hist in list(itertools.product((C, D), repeat=4))]
+
+        self._sixteen_vector = dict(zip(states, sixteen_vector)) # type: Dict[tuple, float]
+        self.classifier['stochastic'] = any(0 < x < 1 for x in set(sixteen_vector))
+
+    def strategy(self, opponent: Player) -> Action:
+        if len(opponent.history) <= 1:
+            return self._initial
+        # Determine which probability to use
+        p = self._sixteen_vector[(tuple(self.history[-2:]),
+                                  tuple(opponent.history[-2:]))]
+        # Draw a random number in [0, 1] to decide
+        return random_choice(p)
+
+
+class AON2(MemoryTwoPlayer):
+    """
+    AON2 a memory two strategy introduced in [Hilbe2017]_. It belongs to the
+    AONk (all-or-none) family of strategies. These strategies were designed to
+    satisfy the three following properties:
+
+    1. Mutually Cooperative. A strategy is mutually cooperative if there are
+    histories for which the strategy prescribes to cooperate, and if it continues
+    to cooperate after rounds with mutual cooperation (provided the last k actions
+    of the focal player were actually consistent).
+
+    2.Error correcting. A strategy is error correcting after at most k rounds if,
+    after any history, it generally takes a group of players at most k + 1 rounds
+    to re establish mutual cooperation.
+
+    3. Retaliating. A strategy is retaliating for at least k rounds if, after
+    rounds in which the focal player cooperated while at least one coplayer defected,
+    the strategy responds by defecting the following k rounds.
+
+    In [Hilbe2017]_ the following vectors are reported as "equivalent" to AON2
+    with their respective self cooperation rate (note that these a not the same):
+    
+    1. [1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1], self cooperation
+    rate: 0.952
+    2. [1, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1], self cooperation
+    rate: 0.951
+    3. [1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1], self cooperation
+    rate:  0.951
+    4. [1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1], self cooperation
+    rate: 0.952
+
+    AON2 is implemented using vector 1 due it's self cooperation rate.
+
+    In essence is a strategy that starts of by cooperating and will cooperate again
+    only after the states (CC, CC), (CD, CD), (DC, DC), (DD, DD).
+
+    Names:
+
+    - AON2: [Hilbe2017]_
+    """
+
+    name = 'AON2'
+    classifier = {
+        'memory_depth': 2,  # Memory-one Four-Vector
```

Comment:
16 vector

Reply:
I agree, I don't think the comment is useful here. :+1: